worker/docker.go

```diff
@@ -97,6 +97,33 @@ type DockerManager struct {
 	mu         *sync.Mutex
 }
 
+func overridePipelineImages(defaultImage string) error {
```

Comment:
@RUFFY-369 could you add a basic function docsting and tests?

Reply:
@rickstaa Thanks for the review! I just added the docstring and the test in the latest commits. :+1: :rocket: